libbpf-tools/sysinjector.bpf.c

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright 2021 Sony Group Corporation */
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include "sysinjector.h"
+
+const volatile pid_t targ_tgid = 0;
+const volatile char targ_comm[TASK_COMM_LEN] = {};
+const volatile int retval = 0;
+
+static __always_inline int comm_allowed(const char *comm)
+{
+	int i;
+
+	for (i = 0; i < TASK_COMM_LEN && targ_comm[i] != '\0'; i++) {
```

Comment:
This will treat "abc" == "abcd", right ?

Reply:
Yes.
I think it is better to match the original implementation of comm_allowed() because we unify the behavior of all tools.
But if you and @ethercflow and maintainers don't mind, I'll try to modify comm_allowed() and move it to _trace_helpers.c_ in the another PR.

Reply:
SGTM